assets/js/page.js

```diff
@@ -0,0 +1,12 @@
+function setPermalinks () {
+  var headers = document.querySelectorAll('.tm-container h4')
+
+  // Apply permalinks to headers
+  if (headers.length > 0) {
+    Array.prototype.forEach.call(headers, function (el, i) {
+      el.innerHTML = '<a href="#' + el.id + '">' + el.innerHTML + '</a>'
+    })
+  }
+}
+
+setPermalinks()
```

Comment:
next time RTFM before building stuff ...

jekyll uses a number of markdown generators, and all of them have this feature built in... 

by default it's on `karmadown`, which can enable this with a flag in `_config.yml` http://kramdown.gettalong.org/converter/html.html